rust/main/chains/hyperlane-sealevel/src/rpc/client.rs

```diff
@@ -245,19 +320,174 @@ impl SealevelRpcClient {
         Ok(None)
     }
 
-    async fn simulate_transaction(
+    pub async fn simulate_transaction(
         &self,
         transaction: &Transaction,
-    ) -> ChainResult<Option<UiTransactionReturnData>> {
-        let return_data = self
+    ) -> ChainResult<RpcSimulateTransactionResult> {
+        let result = self
             .0
-            .simulate_transaction(transaction)
+            .simulate_transaction_with_config(
+                transaction,
+                RpcSimulateTransactionConfig {
+                    sig_verify: false,
+                    replace_recent_blockhash: true,
+                    ..Default::default()
+                },
+            )
             .await
             .map_err(ChainCommunicationError::from_other)?
-            .value
-            .return_data;
+            .value;
+
+        Ok(result)
+    }
+
+    /// Gets the estimated costs for a given instruction.
+    pub async fn get_estimated_costs_for_instruction(
```

Comment:
honestly not sure if these fns should live here, in provider.rs, or somewhere else - curious if anyone has strong takes